include/NeoFOAM/primitives/label.hpp

```diff
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 NeoFOAM authors
+#pragma once
+
+#include <cstdint>
+namespace NeoFOAM
+{
+typedef int32_t label;
```

Comment:
We can switch the size at compile time. We can also use that for localIdx. This might not be super useful for labels but would be good for scalars (NEOFOAM_DP_SCALAR)
```suggestion
#ifdef NEOFOAM_DP_LABELS
    typedef int64_t label;
#else
    typedef int32_t label;
#endif
```

Reply:
We would have one mesh per mpi rank and therefore it can be sufficient to be more specific about the size of the int

Reply:
I agree, this only give the possibility to change the label size at compile time. We still should have a `global_index type ` which is always 64bit.

Reply:
In any case I wouldn't hardcode the label size.

Reply:
changed hardcoded integers to localIdx
Note: There are files which still use int. This should be tackled in a seperate PR